apps/ewallet/lib/ewallet/validators/transaction_consumption_validator.ex

```diff
@@ -222,6 +224,39 @@ defmodule EWallet.TransactionConsumptionValidator do
     end
   end
 
+  def validate_max_consumptions_per_interval(request) do
+    with max <- request.max_consumptions_per_interval,
+         duration <- request.consumption_interval_duration,
+         {false, false} <- {is_nil(max), is_nil(duration)},
+         last_interval_end <- NaiveDateTime.add(NaiveDateTime.utc_now(), -duration, :millisecond),
+         last_consumptions <-
+           TransactionConsumption.get_last_confirmed_consumptions(request.uuid, last_interval_end),
```

Comment:
Confirmations will take longer once we are on-chain. It’ll be possible to blast a lot of pending consumptions pass the limit.

Should we take pending consumptions into the limit counting too?

Reply:
Since it is rechecked at the confirmation step (right before the transaction is made), we don't need to - only one will be confirmed and processed in the end.